drafts/arcalinea-tx-expiry/draft.rst

```diff
@@ -0,0 +1,75 @@
+**Title:** Transaction Expiry
+
+**Author:** Jay Graber
+
+**Status:** Active
+
+**Category:** Standards
+
+**Created:** 2018-01-09
+
+Abstract
+===========
+
+This is an Standards ZIP describing a new consensus rule to set an expiration time for transactions that are not mined to be removed from the mempool.
+
+Motivation
+===========
+
+Transactions that are too large or have insufficient fees are often not mined. This indeterminism is a source of confusion for users and wallets. Allowing transactions to set a block height after which it expires from the mempool would provide certainty around how long a transaction has to confirm before it is rejected by the network and must be re-sent.
+
+Advantages include improving performance by removing transactions unlikely to be mined, and potentially simplifying bidirectional payment channels by reducing the need to store and compress revocations for past states, since transactions not committed to the chain could expire and become invalid after a period of time.
+
+Specification
+===============
+
+Transactions will have a new field, nBlockExpiry, which will set the block height after which transactions will be removed from the mempool if they have not been mined.
+
+The data type for nBlockExpiry will be uint32_t, conforming to the structure of nLockTime. If used in combination with nLockTime, both nLockTime and nBlockExpiry must be block heights.
```

Comment:
"conforming to the structure" implies nBlockExpiry will also have the same semantic interpretation i.e. it can store a unix timestamp instead of just a block height. https://en.bitcoin.it/wiki/Protocol_documentation#tx

"If used in combination..." does this mean if nLockTime is ignored by the transaction, that nBlockExpiry can be a unix timestamp?

Reply:
No, can only be a block height. Will clarify this section